packages/core/application/core-application-browser/src/application.ts

```diff
@@ -297,6 +297,8 @@ export type AppDeepLink<Id extends string = string> = {
   navLinkStatus?: AppNavLinkStatus;
   /** Optional flag to determine if the link is searchable in the global search. Defaulting to `true` if `navLinkStatus` is `visible` or omitted */
   searchable?: boolean;
+  /** Optional category to use over the parent app category */
+  category?: AppCategory;
```

Comment:
I'm surprised to this added without any associated code changed and/or tests being added. IIRC there's some code in this package working with `AppDeepLink`, isn't there? Where is the actual nav tree generated from, again?

Do you mind enlightening me here?

Reply:
Hey @pgayvallet thanks for the quick review!

The `AppDeepLink` type is used on this package when [populating the list of deep links](https://github.com/elastic/kibana/blob/main/packages/core/application/core-application-browser-internal/src/application_service.tsx#L481) while registering an app. In this case, the property is an optional addition to the deepLinks configuration that doesn't affect the deep links usage in this package, since here it only happens to register and update the list itself, without using it.

The [CollapsibleNav](https://github.com/elastic/kibana/blob/main/packages/core/chrome/core-chrome-browser-internal/src/ui/header/collapsible_nav.tsx#L112) component consumes the navLinks derived from the registered deep links and creates the nav tree. It doesn't rely on the deepLink optional category, since those links are grouped under the parent group, such as Analytics, Observability etc...

This addition is only used on the `global_search_providers` plugin. I changed the implementation to use the category provided before the parent app ones, and I tested this behaviour on that plugin.

Hope this helps, let me know if there is anything that you want me to doublecheck or any test that you feel need to exist in the core-application-browser plugin.

Reply:
> The CollapsibleNav component consumes the navLinks derived from the registered deep links and creates the nav tree. It doesn't rely on the deepLink optional category, since those links are grouped under the parent group

Yeah, this is actually what bothers me. You're adding a property to a structure used for multiple features, and only implementing the behavioral change for one of the feature. This is somewhat misleading.

I would at least make that explicit by adding some comment on the property, stating that setting this optional property won't change the way the entry is displayed in the nav, or something